### PR TITLE
Implement Add Amenities (Wifi)

### PR DESCRIPTION
**Pull request by limdanqi**
Implemented the amenities field of study spot, with wifi being the only amenity type now.

Test methods and documentation are not yet included.

Part of #27

### CODE REVIEW

**Comment by limdanqi** [comment]
- Implemented test methods for the Amenity class
- Included documentation for the Amenity class

Resolves #27

File: src/test/java/seedu/address/model/amenity/AmenityTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        assertThrows(IllegalArgumentException.class, () -> new Amenity(invalidAmenityType));
+    }
+
+    @Test
+    public void isValidAmenityType() {
+        // null amenity type
+        assertThrows(NullPointerException.class, () -> Amenity.isValidAmenityType(null));
+
+        // invalid amenity type
+        assertFalse(Amenity.isValidAmenityType("#wifi"));
```
Comment (qreoct) [inline src/test/java/seedu/address/model/amenity/AmenityTest.java:28]: Same issue as `JsonAdaptedStudySpotTest.java` where it looks like we are testing for non-alphanumeric characters, when instead we are testing for Amenities not present in our `Amenity.VALID_TYPES` list.

File: src/test/java/seedu/address/model/amenity/AmenityTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        // one amenity type
+        String[] singleAmenityType = {"wifi"};
+        expectedOutcome = "wifi";
+
+        assertEquals(expectedOutcome, Amenity.listAllAmenityTypes(singleAmenityType));
+
+        // multiple amenity types
+        String[] multipleAmenityTypes = {"wifi", "charger", "food", "air-con"};
+        expectedOutcome = "wifi, charger, food, air-con";
+
```
Comment (qreoct) [inline src/test/java/seedu/address/model/amenity/AmenityTest.java:44]: Would prefer `aircon` instead of `air-con`

**Review by qreoct — CHANGES_REQUESTED** [review]
Overall looks great!

But will need to work with EditCommand before we can merge. Right now Edit still cannot handle Amenities. A suggestion is to find `PREFIX_TAG` in `EditCommandParser.java` use that to build the edit amenity case (which means more testcases can be written for Edit too... but if that's not the scope of this PR it's fine.)

File: src/test/java/seedu/address/storage/JsonAdaptedStudySpotTest.java [hunk 3]
```diff
@@ -23,6 +23,7 @@
     private static final String INVALID_ADDRESS = " ";
     private static final String INVALID_EMAIL = "example.com";
     private static final String INVALID_TAG = "#friend";
+    private static final String INVALID_AMENITY = "#wifi";
```
[the remark this answers is not in the document]
Comment (limdanqi) [reply src/test/java/seedu/address/storage/JsonAdaptedStudySpotTest.java:26]: Yeah i think that makes better sense!

File: src/test/java/seedu/address/storage/JsonSerializableStudyTrackerTest.java [hunk 4]
```diff
@@ -21,14 +18,14 @@
...
-        assertEquals(studyTrackerFromFile, typicalStudySpotsStudyTracker);
-    }
+    //    @Test
+    //    public void toModelType_typicalStudySpotsFile_success() throws Exception {
+    //        JsonSerializableStudyTracker dataFromFile = JsonUtil.readJsonFile(TYPICAL_STUDYSPOTS_FILE,
+    //                JsonSerializableStudyTracker.class).get();
+    //        StudyTracker studyTrackerFromFile = dataFromFile.toModelType();
+    //        StudyTracker typicalStudySpotsStudyTracker = TypicalStudySpots.getTypicalStudyTracker();
+    //        assertEquals(studyTrackerFromFile, typicalStudySpotsStudyTracker);  //todo
+    //    }
```
Comment (qreoct) [inline src/test/java/seedu/address/storage/JsonSerializableStudyTrackerTest.java:28]: Test for future review

File: src/test/java/seedu/address/logic/parser/StudyTrackerParserTest.java [hunk 5]
```diff
@@ -54,14 +50,15 @@ public void parseCommand_delete() throws Exception {
...
-    }
+    //    @Test
+    //    public void parseCommand_edit() throws Exception {
+    //        StudySpot studySpot = new StudySpotBuilder().withName("Test").build();
+    //        EditStudySpotDescriptor descriptor = new EditStudySpotDescriptorBuilder(studySpot).build();
+    //        EditCommand command = (EditCommand) parser.parseCommand(EditCommand.COMMAND_WORD + " "
+    //                + "spot/Test" + " " + StudySpotUtil.getEditStudySpotDescriptorDetails(descriptor));
+    //        assertEquals(new EditCommand(new Name("Test"), descriptor), command);
+    //        //todo discrepancies in amenities and addedamenities
+    //    }
```
Comment (qreoct) [inline src/test/java/seedu/address/logic/parser/StudyTrackerParserTest.java:61]: Test for future review

**Review by qreoct — APPROVED** [review]
Nice!

**Pull request merged**